duplex_http_call.go

```diff
@@ -52,19 +53,22 @@ type duplexHTTPCall struct {
 func newDuplexHTTPCall(
 	ctx context.Context,
 	httpClient HTTPClient,
-	url string,
+	url *url.URL,
 	spec Spec,
 	header http.Header,
 ) *duplexHTTPCall {
 	pipeReader, pipeWriter := io.Pipe()
-	request, err := http.NewRequestWithContext(
-		ctx,
-		http.MethodPost,
-		url,
-		pipeReader,
-	)
-	request.Header = header
-	client := &duplexHTTPCall{
+	request := (&http.Request{
+		Method:     http.MethodPost,
+		URL:        url,
+		Header:     header,
+		Proto:      "HTTP/1.1",
+		ProtoMajor: 1,
+		ProtoMinor: 1,
```

Comment:
This is mirroring what's expected in stdlib. I hate it, but it is what it is. Happy to comment this up, but we can really skip any of the helpers and conveniences that `NewRequestWithContext` provides since our inputs are well known.

Reply:
Makes sense. This is fine with me, but I would like a few comments and a link the the implementation of `NewRequestWithContext`. (The hard-coded HTTP/1.1 constants are offputting to read!)